### PR TITLE
Add Support for Github Actions workflows

### PR DESCRIPTION
**Pull request by xRowe**
1. Add Support for Github Actions workflows
2. When each push or PR, there will start a Action running all the test.
3. Migrate code coverage reporting from Codecov to Coveralls(Need Maintainer further support, like, ReadMe...Coveralls Permission Right...)
4. encode method modified, considering "str" value as physical value.
5. Move all the test files to root path /test/files/
6. Update Test files to satifys workflows
7. etc.....

### CODE REVIEW

File: src/canmatrix/canmatrix.py [hunk 1]
```diff
@@ -436,25 +436,27 @@ def phys2raw(self, value=None):
...
-                )
 
-        if not (self.min <= value <= self.max):
-            logger.info(
+        try:
+            value = decimal.Decimal(value)
+        except Exception as e:
+            raise e
+
+        if not (0 <= value <= 10):
```
Comment (ebroecker) [inline src/canmatrix/canmatrix.py:450]: Hi @xRowe , why 0 to 10 and not self.min and self.max?

File: src/canmatrix/cli/compare.py [hunk 2]
```diff
@@ -23,20 +23,22 @@
...
+
+import canmatrix.log as canmatrix_log
+
 
 logger = logging.getLogger(__name__)
 
 
 @click.command()
-@click.option('-v', '--verbose', 'verbosity', help="Output verbosity", count=True, default=1)
+@click.option('-v', '--verbose', 'verbosity', help="Output verbosity", count=True, default=False)
```
Comment (ebroecker) [inline src/canmatrix/cli/compare.py:41]: is this working ? 

a count-option starting with "False"?

File: src/canmatrix/cli/compare.py [hunk 3]
```diff
@@ -51,23 +53,29 @@ def cli_compare(matrix1, matrix2, verbosity, silent, check_comments, check_attri
...
         verbosity = -1
-    canmatrix.log.set_log_level(root_logger, verbosity)
+    elif not verbosity:
+        # Info Message
+        verbosity = 1
+    else:
+        # Debug Message
+        verbosity = 2
+
+    canmatrix_log.set_log_level(root_logger, verbosity)
```
Comment (ebroecker) [inline src/canmatrix/cli/compare.py:68]: we had (theoretically) multiple verbosity levels, now we only have one? Is that what we want?

File: src/canmatrix/cli/compare.py [hunk 4]
```diff
@@ -51,23 +53,29 @@ def cli_compare(matrix1, matrix2, verbosity, silent, check_comments, check_attri
...
 
     # import only after setting log level, to also disable warning messages in silent mode.
-    import canmatrix.formats  # due this import we need the import alias for log module
+    import canmatrix.formats as canmatrix_format # due this import we need the import alias for log module
```
Comment (ebroecker) [inline src/canmatrix/cli/compare.py:71]: what is the benefit of importing with an alias-name?

File: src/canmatrix/formats/arxml.py [hunk 5]
```diff
@@ -1600,23 +1599,26 @@ def get_frame(frame_triggering, ea, multiplex_translation, float_factory, header
 
         secOC_properties = None
         if pdu is not None and 'SECURED-I-PDU' in pdu.tag:
-            payload_length = ea.get_child(pdu, "LENGTH").text
+            try:
```
Comment (ebroecker) [inline src/canmatrix/formats/arxml.py:1602]: this would fix #799 - perfect

**Review by ebroecker — COMMENTED** [review]
Hi @xRowe 

thank you for you work!

I have some questions added in this review. Maybe I don't understand everything just right.

Additionally:
Why don't you like the "here" in the test-scripts and use paths instead? 

Thanks again

File: src/canmatrix/canmatrix.py (hunk 1 above)
Comment (xRowe) [reply src/canmatrix/canmatrix.py:450]: Test Code...
Found Issue. To be revert to normal code

File: src/canmatrix/cli/compare.py (hunk 2 above)
Comment (xRowe) [reply src/canmatrix/cli/compare.py:41]: Sorry, this will be revert.
Not using count-option before

File: src/canmatrix/cli/compare.py (hunk 3 above)
Comment (xRowe) [reply src/canmatrix/cli/compare.py:68]: Sorry, this will be revert too.

File: src/canmatrix/cli/compare.py (hunk 4 above)
Comment (xRowe) [reply src/canmatrix/cli/compare.py:71]: Yep, Actually not usally here. During the test, I just found the import does not work, so I tried to use as alias
Change can be reverted.

**Comment by xRowe** [comment]
Q: Why don't you like the "here" in the test-scripts and use paths instead?
A: Actually, I never use the PathLib before just want to use a directly way, do not know if it is the best way. Also During the test, I want to create the test file to a temp folder instead of a test folder

**Comment by xRowe** [comment]
@ebroecker 
The Files should be Updated.
Could you please check again?

**Comment by ebroecker** [comment]
Hi @xRowe

thank you for you work, it seems good now!

**Pull request merged**